### PR TITLE
Upgrade to RN 0.50.1

### PR DESCRIPTION
**Pull request by dickie81**
This PR updates times-components to RN 0.50.1

I have manually tested storybook across all platforms and it LGTM.

### CODE REVIEW

**Comment by dickie81** [comment]
Large issue discovered:

Seems we have quite a few components where images have children. This has been deprecated and will need to be rethought.

screen shot 2017-11-07 at 22 02 41

**Comment by dickie81** [comment]
Fixed issue with images. I think this is ready for review / merge.

File: packages/article-flag/package.json [hunk 1]
```diff
@@ -55,20 +55,20 @@
...
     "react-dom": "16.0.0",
-    "react-native": "0.49.3",
+    "react-native": "0.50.1",
     "react-test-renderer": "16.0.0",
     "webpack": "3.3.0"
   },
   "dependencies": {
     "prop-types": "15.6.0",
-    "react-native-svg": "5.4.1",
+    "react-native-svg": "5.5.0",
```
Comment (jimhunty) [inline packages/article-flag/package.json]: Has this been tested?

Comment (dickie81) [reply packages/article-flag/package.json]: SVGs still render on all platforms. According to  react-native-svg's documentation RN 0.50 require's this update.

https://www.npmjs.com/package/react-native-svg

File: packages/author-head/__tests__/__snapshots__/author-head.test.js.snap [hunk 2]
```diff
@@ -250,12 +242,11 @@ exports[`renders with data 1`] = `
               "flex": 1,
             },
             Object {
-              "borderColor": "#FFF",
-              "borderRadius": 50,
-              "borderWidth": 5,
-              "height": 100,
-              "overflow": "hidden",
-              "width": 100,
+              "height": "100%",
```
Comment (jimhunty) [inline packages/author-head/__tests__/__snapshots__/author-head.test.js.snap]: Seems a significant change on this block?

**Comment by killanaca** [comment]
@dickie81 `.flowconfig` should be updated also

**Comment by killanaca** [comment]
and `xcode assets` also to reduce the noise on the next update

File: packages/author-head/__tests__/__snapshots__/author-head.test.js.snap (hunk 2 above)
Comment (dickie81) [reply packages/author-head/__tests__/__snapshots__/author-head.test.js.snap]: Its all related to the issue discovered above. Means I've had to significantly alter the image component:

https://github.com/newsuk/times-components/pull/366/files/88c2e09517902831002c592faf7f0642afa33ebf#diff-78ea4c4e39449a548b4b17c26f052ff3

**Comment by dickie81** [comment]
Might it be worth not squashing this merge to keep the image component update as a separate commit?

File: packages/image/image.js [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
-import { Image } from "react-native";
+import { Image, StyleSheet, View } from "react-native";
 import Placeholder from "./placeholder";
 import imagePropTypes from "./image-prop-types";
 
 const addMissingProtocol = uri => (uri.startsWith("//") ? `https:${uri}` : uri);
 
+const styles = StyleSheet.create({
+  placeholder: {
+    position: "absolute",
```
Comment (sericaia) [inline packages/image/image.js]: I'm a bit afraid of this position absolute rule. I'll have to test ;)

File: packages/brightcove-video/android/build.gradle [hunk 4]
```diff
@@ -25,6 +25,6 @@ repositories {
 }
 
 dependencies {
-    implementation "com.facebook.react:react-native:0.49.+"
+    implementation "com.facebook.react:react-native:0.50.+"
```
Comment (sericaia) [inline packages/brightcove-video/android/build.gradle]: I'm getting this warning:

“To suppress this warning, remove “buildToolsVersion ‘23.0.1’” from your build.gradle file, as each version of the Android Gradle Plugin now has a default version of the build tools. "

Shouldn't we change this here? :)

Comment (sericaia) [reply packages/brightcove-video/android/build.gradle]: its fine as long as you update on Android Studio 3

File: packages/image/image.js (hunk 3 above)
Comment (sericaia) [reply packages/image/image.js]: Its fine, I just confirmed

**Pull request merged**